### DISCUSSION TITLE
Push Notification w/ OneSignal

### DISCUSSION THREAD

**Discussion by furkananter** [supabase/supabase#15961] [Questions]
Hi everyone, I am building an application these days. Right now I need to handle the push notification system and I am using One Signal. One Signal is working correctly. I can get and send notifications on dashboard but this app needs to be more complex. 

For example; I have a users and likes table. Users is liking their posts but they can not get any notifications. I need to make that feature quickly.

- When I look out One Signal's documentation and video; they used webhook and edge functions. I deployed this function to my supabase; 
``` import {serve} from 'https://deno.land/std@0.177.0/http/server.ts';
import * as OneSignal from 'https://esm.sh/@onesignal/node-onesignal@1.0.0-beta7';

const _OnesignalAppId_ = Deno.env.get('ONESIGNAL_APP_ID')!;
const _OnesignalUserAuthKey_ = Deno.env.get('USER_AUTH_KEY')!;
const _OnesignalRestApiKey_ = Deno.env.get('ONESIGNAL_REST_API_KEY')!;
const configuration = OneSignal.createConfiguration({
  userKey: _OnesignalUserAuthKey_,
  appKey: _OnesignalRestApiKey_,
});

const onesignal = new OneSignal.DefaultApi(configuration);

serve(async req => {
  try {
    const {record} = await req.json();

    // Build OneSignal notification object
    const notification = new OneSignal.Notification();
    notification.app_id = _OnesignalAppId_;
    notification.include_external_user_ids = [record.user_id];
    notification.contents = {
      en: `$${record.user_id} likes your post!`,
    };
    const onesignalApiRes = await onesignal.createNotification(notification);

    console.log('onesignalapires', onesignalApiRes);
    console.log('onesignalapires', onesignalApiRes.body);

    return new Response(JSON.stringify({onesignalResponse: onesignalApiRes}), {
      headers: {'Content-Type': 'application/json'},
    });
  } catch (err) {
    console.error('Failed to create OneSignal notification', err);
    return new Response('Server error.', {
      headers: {'Content-Type': 'application/json'},
      status: 400,
    });
  }
});
```
Then I am setting a webhook to connects my edge functions which name is notify. Then I look out the logs and i am seeing these logs: 
- onesignalapires undefined 
- onesignalapires e { id: "", errors: e {} }

How can i make this? I think this is so complex. I didn't see anything related with React Native. Flutter, Next, React exists but React Native CLI doesn't.

**Comment by blondie63** [comment]
Hi, have you found a solution ?

**Comment by araymond11** [comment]
Hi @furkananter, I'm doing exactly like you and it fails when calling createNotification. Did you find a solution for that ? 

Error: HTTP-Code: 400
Message: Bad Request
Body: {"errors":[{}]}
